File: Native_Kotlin/app/src/main/java/com/example/activityautostart/selector/SelectorAdapter.kt
package com.example.activityautostart.selector

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.example.activityautostart.data.DataModel
import com.example.movch.activityautostart.R

class SelectorAdapter(private val controller: SelectorController): RecyclerView.Adapter<SelectorHolder>() {
    private var dataList: List<DataModel> = listOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SelectorHolder =
        SelectorHolder(LayoutInflater.from(parent.context)
            .inflate(R.layout.holder_selector_card, parent, false), controller)

    override fun getItemCount(): Int = dataList.size

    override fun onBindViewHolder(holder: SelectorHolder, position: Int) {
        val dataModel = dataList[position]
        holder.init(dataModel)
    }

    fun setAllData(dataList: List<DataModel>) {
        this.dataList = dataList
    }
}

File: Xamarin_CSharp/ActivityAutostart/Recycler/SelectorAdapter.cs
using System;
using System.Collections.Generic;
using Android.Support.V7.Widget;
using Android.Views;
using Portable.Data;

namespace ActivityAutostart.Recycler
{
    public class SelectorAdapter : RecyclerView.Adapter
    {
        public event Action<string> OnSelectorClick;

        private List<DataModel> _dataList;

        public override int ItemCount => _dataList?.Count ?? 0;

        public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int position)
        {
            if (holder is SelectorHolder selectorHolder)
            {
                selectorHolder.OnSelectorClick -= SelectorClicked;
                selectorHolder.Init(_dataList[position]);
                selectorHolder.OnSelectorClick += SelectorClicked;
            }
        }
        
        public override RecyclerView.ViewHolder OnCreateViewHolder(ViewGroup parent, int viewType)
        {
            return new SelectorHolder(LayoutInflater.From(parent.Context).Inflate(Resource.Layout.holder_selector, parent, false));
        }

        public void SetAllData(List<DataModel> dataList)
        {
            _dataList = dataList;
        }

        private void SelectorClicked(string id)
        {
            OnSelectorClick?.Invoke(id);
        }
    }
}